### DISCUSSION TITLE
Problems with detailed routing violations with an array of macros

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#2804] [General]
In this pull request, I've added a design that has a problem with detailed routing.  I *suspect* there are some alignment requirements for macro positioning, but I don't understand what those requirements are.

Any tips?

https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/782

![image](https://user-images.githubusercontent.com/2798822/213920736-10b8a7f8-0b71-466a-9357-178dbb9b52b7.png)

**Comment by maliberty** [comment]
My first guess would be to look at the macro pins and whether they are on grid

**Reply by oharboe** [reply]
What is the grid? Where do I find it's definition? Is there a way to "snap to grid"?

**Reply by maliberty** [reply]
I was thinking of the manufacturing grid defined in your tech lef.  Since you are manually placing your instances its up to you to align them.  I'm not certain this is the problem, just a first guess.

**Reply by oharboe** [reply]
In ```platforms/asap7/lef/asap7_tech_1x_201209.lef```, I find:

```
UNITS
 DATABASE MICRONS 1000 ;
END UNITS
 MANUFACTURINGGRID 0.001 ;
```

There are some placements that are not rounded to the manufacutring grid:

```
ces_15_2 R0 59.400000000000006 372
```

I corrected the code so it is aligned to the manufcaturing grid: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/783/commits/877ec94cd0d165c46ac6790eb9ffe70c498414f5

Still:

```
[INFO DRT-0195] Start 0th optimization iteration.
Failed to find a path between pin ces_5_0/io_outs_2[3] and source aps:
( 382 529 2 ) (Idx) / ( 18081 147420 ) (coords)
extBBox (xDim, yDim, zDim) = (602, 711, 10)
[ERROR DRT-0255] Maze Route cannot find path of net ces_5_0_io_outs_2\[3\] in worker of routeBox ( 15120 143640 ) ( 18900 147420 ).
Failed to find a path between pin  and source aps:
( 195 187 4 ) (Idx) / ( 91236 143650 ) (coords)
extBBox (xDim, yDim, zDim) = (602, 711, 10)
[ERROR DRT-0255] Maze Route cannot find path of net ces_5_3_io_ins_2\[6\] in worker of routeBox ( 90720 143640 ) ( 94500 147420 ).
Failed to find a path between pin  and source aps:
( 279 187 4 ) (Idx) / ( 114996 143650 ) (coords)
extBBox (xDim, yDim, zDim) = (602, 711, 10)
[ERROR DRT-0255] Maze Route cannot find path of net ces_5_4_io_ins_2\[6\] in worker of routeBox ( 113400 143640 ) ( 117180 147420 ).
Error: detail_route.tcl, 47 DRT-0255
Elapsed time: 0:36.67[h:]min:sec. CPU time: user 404.45 sys 75.57 (1309%). Peak memory: 13355620KB.
make: *** [results/asap7/mock-very-large-array/base/5_2_route.odb] Error 1
```

**Comment by khaledhasanzami** [comment]
Same issue, any solution reached yet?

**Reply by oharboe** [reply]
> Same issue, any solution reached yet?

The particular issue in this pull request has been solved.

Can you create a new testcase?

If it is a confidential design, look at deltaDebug.py to whittle down the dedign to see if you could communicate in private to the maintainers.

**Comment by oharboe** [comment]
We're not seeing these problems currently with latest master
